src/commands/ssh.rs

```diff
@@ -0,0 +1,273 @@
+    environment: Option<String>,
+
+    #[clap(short, long)]
+    /// Deployment instance ID to connect to (defaults to first active instance)
+    #[arg(long = "deployment-instance", value_name = "deployment-instance-id")]
+    deployment_instance: Option<String>,
+}
+
+async fn get_ssh_connect_params(
+    args: Args,
+    configs: &Configs,
+    client: &Client,
+) -> Result<SSHConnectParams> {
+    let linked_project = configs.get_linked_project().await?;
+
+    let project_id = args.project.unwrap_or(linked_project.project.clone());
+    let project = get_project(client, configs, project_id.clone()).await?;
+
+    let environment = args
+        .environment
+        .clone()
+        .unwrap_or(linked_project.environment.clone());
+    let environment_id = get_matched_environment(&project, environment)?.id;
+
+    let service_id = get_or_prompt_service(linked_project.clone(), project, args.service)
+        .await?
+        .ok_or_else(|| anyhow::anyhow!("No service found. Please specify a service to connect to via the `--service` flag."))?;
+
+    Ok(SSHConnectParams {
+        project_id,
+        environment_id,
+        service_id,
+        deployment_instance_id: args.deployment_instance,
+    })
+}
+
+#[cfg(unix)]
+async fn setup_signal_handlers() -> Result<(
+    tokio::signal::unix::Signal,
+    tokio::signal::unix::Signal,
+    tokio::signal::unix::Signal,
+)> {
+    let sigint = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::interrupt())?;
+    let sigterm = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate())?;
+    let sigwinch = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::window_change())?;
+
+    Ok((sigint, sigterm, sigwinch))
+}
+
+#[cfg(not(unix))]
+async fn setup_signal_handlers() -> Result<()> {
+    // On Windows, we don't have these Unix signals
+    Ok(())
+}
+
+pub async fn command(args: Args, _json: bool) -> Result<()> {
+    let configs = Configs::new()?;
+    let client = GQLClient::new_authorized(&configs)?;
+
+    let params = get_ssh_connect_params(args, &configs, &client).await?;
+
+    let token = configs
+        .get_railway_auth_token()
+        .context("No authentication token found. Please login first with 'railway login'")?;
+
+    let spinner = indicatif::ProgressBar::new_spinner()
+        .with_style(
+            indicatif::ProgressStyle::default_spinner()
+                .tick_chars(TICK_STRING)
+                .template("{spinner:.green} {msg}")?,
+        )
+        .with_message("Connecting to service...");
+
+    spinner.enable_steady_tick(std::time::Duration::from_millis(100));
+
+    let ws_url = format!("wss://{}", configs.get_relay_host_path());
+
+    let mut client = TerminalClient::new(&ws_url, &token, &params).await?;
+
+    // Wait a moment before getting terminal size and sending events to connection
+    tokio::time::sleep(std::time::Duration::from_millis(300)).await;
```

Comment:
I added this because for some reason the terminal size isn't correct or the backend isn't ready yet for the send_window_size event. This allows initial window size to correctly be set on connect.